### PR TITLE
Disruptor native support

### LINKED ISSUE #1597: Disruptor native support

### PR DESCRIPTION
**Pull request by lburgazzoli**
Fixes #1597

### CODE REVIEW

**Review by oscerd — APPROVED** [review]

File: extensions/disruptor/runtime/pom.xml [hunk 1]
```diff
@@ -57,6 +57,11 @@
             <groupId>org.apache.camel</groupId>
             <artifactId>camel-disruptor</artifactId>
         </dependency>
+        <dependency>
+            <groupId>org.graalvm.nativeimage</groupId>
+            <artifactId>svm</artifactId>
+            <scope>provided</scope>        </dependency>
```
Comment (davsclaus) [inline extensions/disruptor/runtime/pom.xml:63]: Tidy up

File: extensions/disruptor/runtime/pom.xml [hunk 2]
```diff
@@ -87,6 +92,8 @@
                             <version>${quarkus.version}</version>
                         </path>
                     </annotationProcessorPaths>
+                    <source>9</source>
```
Comment (davsclaus) [inline extensions/disruptor/runtime/pom.xml:95]: Should this not be 11 instead of 9 if we go higher than 8

File: extensions/disruptor/runtime/src/main/resources/META-INF/quarkus-extension.yaml [hunk 3]
```diff
@@ -24,9 +24,8 @@
 name: "Camel Disruptor"
 description: "Provides asynchronous SEDA behavior using LMAX Disruptor"
 metadata:
-  unlisted: true
   guide: "https://camel.apache.org/camel-quarkus/latest/reference/extensions/disruptor.html"
   categories:
   - "integration"
   status:
-  - "preview"
+  - "stable"
```
Comment (davsclaus) [inline extensions/disruptor/runtime/src/main/resources/META-INF/quarkus-extension.yaml:31]: Is it really stable when we just landed native support?

File: docs/modules/ROOT/pages/reference/extensions/disruptor.adoc [hunk 4]
```diff
@@ -28,3 +28,8 @@ Please refer to the above links for usage and configuration details.
 ----
 
 Check the xref:user-guide/index.adoc[User guide] for more information about writing Camel Quarkus applications.
+
+== Camel Quarkus limitations
+
+The disruptor-vm provide support for communication across CamelContext instances but since camel-quarkus support a single CamelContext, such component is not available.
```
Comment (ppalaga) [inline docs/modules/ROOT/pages/reference/extensions/disruptor.adoc:34]:
```suggestion
The `disruptor-vm` component is not available on Camel Quarkus. This is because it is supposed to provide support for communication across multiple CamelContext instances within a single JVM, but by design, there is always just a single `CamelContext` on Camel Quarkus. Therefore `disruptor-vm` would make no sense.
```

File: extensions/disruptor/deployment/src/main/java/org/apache/camel/quarkus/component/disruptor/deployment/DisruptorProcessor.java [hunk 5]
```diff
@@ -0,0 +1,75 @@
...
+        //   Warning: RecomputeFieldValue.ArrayBaseOffset automatic substitution failed. The automatic substitution
+        //   registration was attempted because a call to jdk.internal.misc.Unsafe.arrayBaseOffset(Class) was detected
+        //   in the static initializer of com.lmax.disruptor.RingBufferFields. Detailed failure reason(s): Could not
+        //   determine the field where the value produced by the call to jdk.internal.misc.Unsafe.arrayBaseOffset(Class)
+        //   for the array base offset computation is stored. The call is not directly followed by a field store or by
+        //   a sign extend node followed directly by a field store.
+        //
+        // Even if this is reported as a warning and the native compilation succeed, some static field are not computed
+        // properly which result in weird result as runtime. For such reason, the static init method need to re-run at
+        // runtime.
```
Comment (ppalaga) [inline extensions/disruptor/deployment/src/main/java/org/apache/camel/quarkus/component/disruptor/deployment/DisruptorProcessor.java:72]: I appreciate the comment!

File: extensions/disruptor/runtime/pom.xml [hunk 6]
```diff
@@ -57,6 +57,11 @@
             <groupId>org.apache.camel</groupId>
             <artifactId>camel-disruptor</artifactId>
         </dependency>
+        <dependency>
+            <groupId>org.graalvm.nativeimage</groupId>
+            <artifactId>svm</artifactId>
+            <scope>provided</scope>        </dependency>
+
```
Comment (ppalaga) [inline extensions/disruptor/runtime/pom.xml:64]:
```suggestion
            <scope>provided</scope>
        </dependency>

```

File: extensions/disruptor/runtime/pom.xml (hunk 2 above)
Comment (lburgazzoli) [reply extensions/disruptor/runtime/pom.xml:95]: looks like some IDE intentions applied, fixing it

File: extensions/disruptor/runtime/src/main/resources/META-INF/quarkus-extension.yaml (hunk 3 above)
Comment (lburgazzoli) [reply extensions/disruptor/runtime/src/main/resources/META-INF/quarkus-extension.yaml:31]: @ppalaga what do you think ? I don't remember the rationale

File: extensions/disruptor/runtime/pom.xml [hunk 7]
```diff
@@ -87,6 +92,8 @@
                             <version>${quarkus.version}</version>
                         </path>
                     </annotationProcessorPaths>
+                    <source>9</source>
+                    <target>9</target>
```
Comment (ppalaga) [inline extensions/disruptor/runtime/pom.xml:96]: We still support target 1.8. If this is really required, it should be properly explained in the limitations.adoc, IMO. Why is this necessary, BTW?

File: integration-tests/disruptor/pom.xml [hunk 8]
```diff
@@ -0,0 +1,160 @@
...
+            <artifactId>rest-assured</artifactId>
+            <scope>test</scope>
+        </dependency>
+        <dependency>
+            <groupId>org.awaitility</groupId>
+            <artifactId>awaitility</artifactId>
+            <scope>test</scope>
+        </dependency>
+
+        <!-- The following dependencies guarantee that this module is built after them. You can update them by running `mvn process-resources -Pformat -N` from the source tree root directory -->
```
Comment (ppalaga) [inline integration-tests/disruptor/pom.xml:86]: Note to myself: We should perhaps make the regen of these virtual deps a part of the regular build. Otherwise ppl. won't regen them and the CI will fail with strange dependency errors. And yes, the vitual deps are out of sync here.

**Review by ppalaga — COMMENTED** [review]
Thanks, some comments inline.

File: extensions/disruptor/runtime/src/main/resources/META-INF/quarkus-extension.yaml (hunk 3 above)
Comment (ppalaga) [reply extensions/disruptor/runtime/src/main/resources/META-INF/quarkus-extension.yaml:31]: We currently consider all extensions stable that are native and have a proper test.

File: extensions/disruptor/runtime/pom.xml (hunk 7 above)
Comment (lburgazzoli) [reply extensions/disruptor/runtime/pom.xml:96]: Already spotted by @davsclaus

File: extensions/disruptor/runtime/pom.xml (hunk 6 above)
Comment (lburgazzoli) [reply extensions/disruptor/runtime/pom.xml:64]: Already spotted by @davsclaus

**Comment by lburgazzoli** [comment]
@davsclaus @ppalaga fixed findings, except the preview vs stable as I don't know what should be the solution

File: integration-tests/disruptor/pom.xml (hunk 8 above)
Comment (lburgazzoli) [reply integration-tests/disruptor/pom.xml:86]: anything I need to do ?

Comment (ppalaga) [reply integration-tests/disruptor/pom.xml:86]: `mvn process-resources -Pformat -N` from the top folder should suffice

Comment (lburgazzoli) [reply integration-tests/disruptor/pom.xml:86]: don't see anything changed

Comment (ppalaga) [reply integration-tests/disruptor/pom.xml:86]: I am blind, sorry, the deps. I was expecting to see as virtual are from `io.quarkus` but of course, those do not need any virtual counterpart. All is fine, sorry again for the false alarm!

**Review by ppalaga — APPROVED** [review]

File: integration-tests/disruptor/pom.xml (hunk 8 above)
Comment (lburgazzoli) [reply integration-tests/disruptor/pom.xml:86]: no worries :)
